tests/app/UnitTests/GitUI.Tests/Editor/Diff/DiffHighlightServiceTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Castle.Core;
+using FluentAssertions;
+using GitUI.Editor.Diff;
+using ICSharpCode.TextEditor.Document;
+
+namespace GitUITests.Editor.Diff;
+
+[TestFixture]
+public class DiffHighlightServiceTests
+{
+    [Test]
+    public void GetDifferenceMarkers()
+    {
+        // LineSegment is hard to create. Use TextMarker as implementation type of ISegment for this test.
+        const TextMarkerType dontCare = TextMarkerType.SolidBlock;
+
+        const string identicalPartBefore = "identical_part_before_";
+        const string identicalPartAfter = "_identical_part_after";
+        const string differentRemoved = "RemovedX";
+        const string differentAdded = "AddedY";
+        const string removedLineText = $"-{identicalPartBefore}{differentRemoved}{identicalPartAfter}";
+        const string addedLineText = $"+{identicalPartBefore}{differentAdded}{identicalPartAfter}";
+        const string text = $"{removedLineText}\n{addedLineText}";
+        TextMarker removedLine = new(offset: text.IndexOf(removedLineText), removedLineText.Length, textMarkerType: dontCare);
+        TextMarker addedLine = new(offset: text.IndexOf(addedLineText), addedLineText.Length, textMarkerType: dontCare);
+        const int lineStartOffset = 1;
+
+        IEnumerable<TextMarker> markers = DiffHighlightService.GetDifferenceMarkers(GetCharAt, removedLine, addedLine, lineStartOffset);
```

Comment:
I had planned to test the refactored GetBlockOfLines() in my branch, the test for "merged gaps" can be done more easily.
That will make more extensive tests possible as a compliment.
(Combined use that use not adjacent blocks will require tests like this too.)